### PR TITLE
Update tplink documentation for next release

### PR DESCRIPTION
**Pull request by sdb9696**
## Proposed change

Update tplink docs for next release

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:  https://github.com/home-assistant/core/pull/120060
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
  - Expanded support for new device categories in the `tplink_tapo` integration, including Binary Sensor, Button, Climate, Fan, Number, and Select.
  - Updated light effects support, including new details for Kasa bulbs and services for setting random and sequence effects on light strips.

- **Documentation**
  - Revised supported device listings to refer to external sources for the most current information.
  - Included footnotes for devices requiring authentication and cross-brand hub compatibility.

### CODE REVIEW

**Comment by rytilahti** [comment]
@sdb9696 when we are that far, please sync the platforms into this one single PR as it makes no sense to do what I just did earlier... :-)

File: source/_integrations/tplink.markdown [hunk 1]
```diff
@@ -2,6 +2,7 @@
 title: TP-Link Smart Home
 description: Instructions on integrating TP-Link Smart Home Devices to Home Assistant.
 ha_category:
```
Comment (rytilahti) [inline source/_integrations/tplink.markdown:4]: When you update this, don't forget to update the `tplink_tapo.markdown`, too.

File: source/_integrations/tplink.markdown [hunk 2]
```diff
@@ -36,98 +38,33 @@ If you have an older device that does not currently require authentication, you
...
+- **Power Strips**: P300, TP25
+- **Wall Switches**: S500D, S505, S505D
+- **Bulbs**: L510B, L510E, L530E
+- **Light Strips**: L900-10, L900-5, L920-5, L930-5
+- **Hubs**: H100
+- **Hub-Connected Devices<sup>\*\*\*</sup>**: T110, T300, T310, T315
+
+<sup>\*</sup>&nbsp;&nbsp; Model requires authentication<br>
+<sup>\*\*</sup>&nbsp; Newer versions require authentication<br>
+<sup>\*\*\*</sup> Devices may work across TAPO/KASA branded hubs
```
Comment (bdraco) [inline source/_integrations/tplink.markdown]: Ideally we keep markdown syntax and avoid the `&nbsp;` and `<sup>`

File: source/_integrations/tplink.markdown (hunk 1 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown:4]: Done

File: source/_integrations/tplink.markdown (hunk 2 above)
Comment (sdb9696) [reply source/_integrations/tplink.markdown]: Fixed

**Review by bdraco — DISMISSED** [review]

**Review by bdraco — APPROVED** [review]

**Pull request merged**